homeassistant/helpers/area_registry.py

```diff
@@ -0,0 +1,134 @@
+"""Provide a way to connect devices to one physical location."""
+import logging
+import uuid
+from collections import OrderedDict
+from typing import Optional
+
+import attr
+
+from homeassistant.core import callback
+from homeassistant.loader import bind_hass
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_REGISTRY = 'area_registry'
+
+STORAGE_KEY = 'core.area_registry'
+STORAGE_VERSION = 1
+SAVE_DELAY = 10
+
+
+@attr.s(slots=True, frozen=True)
+class AreaEntry:
+    """Area Registry Entry."""
+
+    name = attr.ib(type=str, default=None)
+    id = attr.ib(type=str, default=attr.Factory(lambda: uuid.uuid4().hex))
+
+
+class AreaRegistry:
+    """Class to hold a registry of areas."""
+
+    def __init__(self, hass) -> None:
+        """Initialize the area registry."""
+        self.hass = hass
+        self.areas = None
+        self._store = hass.helpers.storage.Store(STORAGE_VERSION, STORAGE_KEY)
+
+    @callback
+    def async_create(self, name: str):
+        """Create a new area."""
+        if self._async_is_registered(name):
+            raise ValueError('Name is already in use')
+
+        area = AreaEntry()
+        self.areas[area.id] = area
+
+        return self.async_update(area.id, name=name)
+
+    @callback
+    def async_delete(self, area_id: str) -> None:
+        """Delete area."""
+        del self.areas[area_id]
+
+    @callback
+    def async_read(self) -> AreaEntry:
```

Comment:
We haven't used `read` in other places. For auth we use `async_get_users`, so maybe async_get_areas ?